client/delegate-auth/delegate-auth-script.ts

```diff
@@ -0,0 +1,163 @@
+    });
+    setCacheClientOptions(chainId, {
+        url: cacheServerUrl
+    });
+
+    setChainConfig(chainId, {
+        rpcUrl
+    });
+    
+    const connectionInfos = await iamAgent.initializeConnection({
+        walletProvider: WalletProvider.WalletConnect,
+        initCacheServer: true,
+        createDocument: true,
+    });
+    console.log(`${emoji.get('large_green_circle')} IAM Owner created \x1b[32m%s\x1b[0m`, `\n`);
+    console.log(`\t${emoji.get('scroll')} owner DID: \x1b[32m%s\x1b[0m`, `${connectionInfos.did}\n`);
+    console.log(`\t${emoji.get('scroll')} owner identity token: \x1b[32m%s\x1b[0m`, `${connectionInfos.identityToken}\n`);
+    return iamAgent;
+};
+
+const registerAsset = async (assetOwner: IAM): Promise<Partial<{ assetAddress: string, assetPubKey: string, assetPrivKey: string }>> => {
+
+    // Creates a new asset using iam-client-lib registerAsset()     
+    const assetAddress: string = await assetOwner.registerAsset();
+    console.log(`${emoji.get('large_green_circle')} Asset Registeration:\n`);
+    console.log(`\t${emoji.get('hammer_and_wrench')}  Asset Created at address: \x1b[32m%s\x1b[0m`, `${assetAddress}\n`);
+
+    // // Generates a new secp256k1 keypair
+    const keyPair = new Keys();
+    console.log(`\t${emoji.get('closed_lock_with_key')} Setting asset's Keypair: \x1b[32m%s\x1b[0m`, `\n`);
+    console.log(`\t\t${emoji.get('key')} Private Key: \x1b[32m%s\x1b[0m`, `${keyPair.privateKey}\n`);
+    console.log(`\t\t${emoji.get('lock')} Public Key: \x1b[32m%s\x1b[0m`, `${keyPair.publicKey}\n`);
+
+    return {
+        assetAddress,
+        assetPubKey: keyPair.publicKey,
+        assetPrivKey: keyPair.privateKey,
+    };
+};
+
+const addKeyToAssetDocument = async (assetOwner: IAM, assetAddress: string, assetPublicKey: string): Promise<boolean> => {
+
+    const assetDid = `did:ethr:${assetAddress}`;
+    const didOptions = {
+        didAttribute: DIDAttribute.PublicKey,
+        did: assetDid,
+        data: {
+            algo: Algorithms.Secp256k1,
+            encoding: Encoding.HEX,
+            type: PubKeyType.SignatureAuthentication2018,
+            value: { tag: "key-1", publicKey: `0x${assetPublicKey}` },
+        },
+    }
+    // Add new key to asset's DID Document
+    const isDIdDocUpdated = await assetOwner.updateDidDocument(didOptions);
+
+    return isDIdDocUpdated;
+};
+
+const displayError = (err: Error, step: string) => {
+    console.log(`${emoji.get('red_circle')} [ ${step} ] : \x1b[31m%s\x1b[0m`, `: ${err}\n`);
+};
+
+const connectToBackend = async (iamAgent: IAM, token: string, backendUrl: string) : Promise<string> => {
+    const response = await axios.post<{ token: string }>(
+        `${backendUrl}/login`,
+        {
+            identityToken: token,
+        },
+        { withCredentials: true }
+    );
+    return response.data.token;
+}
+
+const createIdentityProofWithDelegate = async (secp256k1PrivateKey: string, rpcUrl: string, identityProofDid: string) => {
```

Comment:
just a note to replace this function with the new function from iam-client-lib (when it's available)